### PR TITLE
feat: add custom callback example and reference colab

### PR DESCRIPTION
**Pull request by parambharat**
## Description

This PR adds details to the huggingface documentation related to custom logging and callbacks. Additionally, we also fix the description of the `WANDB_LOG_MODEL` env var to reflect the new changes in the HF integration.

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/integrations/huggingface.md [hunk 1]
```diff
@@ -346,6 +346,58 @@ wandb.init(project="amazon_sentiment_analysis",
 ### Custom logging
 
 Logging to Weights & Biases via the [Transformers `Trainer` ](https://huggingface.co/transformers/main\_classes/trainer.html)is taken care of by the `WandbCallback` ([reference documentation](https://huggingface.co/transformers/main\_classes/callback.html#transformers.integrations.WandbCallback)) in the Transformers library. If you need to customize your Hugging Face logging you can modify this callback.
+For instance here's how you could modify the callback to log predictions to a Weights & Biases table after each evaluation
```
Comment (morganmcg1) [inline docs/guides/integrations/huggingface.md]: 1. Can you add a header to this section: “View evaluation samples during training”
2. Can you complete the code example to show the callback being added to the Trainer
3. Show how to set an optional logging frequency if they don’t want to log every eval_steps
4. A bit more explanation of what is going on in this code

**Review by morganmcg1 — CHANGES_REQUESTED** [review]

File: docs/guides/integrations/huggingface.md (hunk 1 above)
Comment (parambharat) [reply docs/guides/integrations/huggingface.md]: @morganmcg1 , Thanks for the detailed review comments. I have addressed all of them. Please let me know what you think and resolve the conversation if you are satisfied.

**Review by morganmcg1 — APPROVED** [review]
LGTM, thanks for the changes!

**Comment by morganmcg1** [comment]
@ngrayluna would love a merge on this whenever you get a chance, as lukas has been asking for it

**Comment by morganmcg1** [comment]
fixed merge conflicts

**Review by ngrayluna — CHANGES_REQUESTED** [review]
Build failing. Markdown format needs fix.

**Pull request merged**